File: fubon.wms/src/webapp/assets/txn/PMS314/PMS314.js
/*
  交易畫面邏輯定義
  請修改 Controller 名稱，以符合交易畫面的定義
 */
'use strict';
eSoafApp.controller('PMS314Controller',
	function($rootScope, $scope, $controller, $confirm, socketService, ngDialog, projInfoService) {
		$controller('BaseController', {$scope: $scope});
		$scope.controllerName = "PMS314Controller";
	
		$scope.init = function(){
			$scope.inputVO = {
					seq: 0,
					summitINC: 0,
					peakINC: 0,
					summitFC1: 'N',
					summitFC2: 'N',
					summitFC3: 'N',
					summitFC4: 'N',
					summitFC5: 'N',
					peakFC1: 'N',
					peakFC2: 'N',
					peakFC3: 'N',
					peakFC4: 'N',
					peakFC5: 'N',
					fch1INC: 0,
					fch2INC: 0,
					fchAum: 0,
					fchNew: 0
			};			                     
		};
		$scope.init();
	    
		$scope.inquire = function(){
			$scope.sendRecv("PMS314", "inquire", "com.systex.jbranch.app.server.fps.pms314.PMS314InputVO", {},
					function(tota, isError) {
						if (!isError) {
							if(tota[0].body.resultList.length == 0) {
								$scope.paramList=[];
								$scope.inputVO.seq = [];
								$scope.inputVO.summitINC = [];
							    $scope.inputVO.peakINC = [];
							    $scope.inputVO.summitFC1 = [];
							    $scope.inputVO.summitFC2 = [];
							    $scope.inputVO.summitFC3 = [];
							    $scope.inputVO.summitFC4 = [];
							    $scope.inputVO.summitFC5 =[];
							    $scope.inputVO.peakFC1 = [];
							    $scope.inputVO.peakFC2 = [];
							    $scope.inputVO.peakFC3 = [];
							    $scope.inputVO.peakFC4 = [];
							    $scope.inputVO.peakFC5 = [];
							    $scope.inputVO.fch1INC =[];
							    $scope.inputVO.fch2INC =[];
							    $scope.inputVO.fchAum = [];
							    $scope.inputVO.fchNew = [];
								$scope.showMsg("ehl_01_common_009");
	                			return;
	                		}
//							$scope.originalList = angular.copy(tota[0].body.resultList);
							$scope.paramList = tota[0].body.resultList;	

							$scope.inputVO.seq = $scope.paramList[0].SEQ;
							$scope.inputVO.summitINC = $scope.paramList[0].FC_SUMMIT_INC;
						    $scope.inputVO.peakINC = $scope.paramList[0].FC_PEAK_INC;
						    $scope.inputVO.summitFC1 = $scope.paramList[0].SUMMIT_FC1;
						    $scope.inputVO.summitFC2 = $scope.paramList[0].SUMMIT_FC2;
						    $scope.inputVO.summitFC3 = $scope.paramList[0].SUMMIT_FC3;
						    $scope.inputVO.summitFC4 = $scope.paramList[0].SUMMIT_FC4;
						    $scope.inputVO.summitFC5 = $scope.paramList[0].SUMMIT_FC5;
						    $scope.inputVO.peakFC1 = $scope.paramList[0].PEAK_FC1;
						    $scope.inputVO.peakFC2 = $scope.paramList[0].PEAK_FC2;
						    $scope.inputVO.peakFC3 = $scope.paramList[0].PEAK_FC3;
						    $scope.inputVO.peakFC4 = $scope.paramList[0].PEAK_FC4;
						    $scope.inputVO.peakFC5 = $scope.paramList[0].PEAK_FC5;
						    $scope.inputVO.fch1INC = $scope.paramList[0].FCH_1_INC;
						    $scope.inputVO.fch2INC = $scope.paramList[0].FCH_2_INC;
						    $scope.inputVO.fchAum = $scope.paramList[0].FCH_AUM;
						    $scope.inputVO.fchNew = $scope.paramList[0].FCH_NEW;
						    
							$scope.outputVO = tota[0].body;	
							
							return;
						}
			});
		};
		$scope.inquire();
		
		/*** 儲存功能 ***/
		$scope.save = function () {	
//			alert("ss="+$scope.inputVO.summitFC1);
			$scope.sendRecv("PMS314", "save", "com.systex.jbranch.app.server.fps.pms314.PMS314InputVO", $scope.inputVO,
					function(tota, isError) {
						if (isError) {
		            		$scope.showErrorMsgInDialog(tota[0].body.msgData);
		            	}
		            	if (tota.length > 0) {
		            		$scope.showMsg('儲存成功');
		            		$scope.inquire();
		            	};		
			});
			
		};
		
});

File: fubon.wms/src/main/java/com/systex/jbranch/app/server/fps/pms314/PMS314.java
package com.systex.jbranch.app.server.fps.pms314;

import java.text.SimpleDateFormat;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import com.systex.jbranch.fubon.commons.FubonWmsBizLogic;
import com.systex.jbranch.platform.common.dataaccess.delegate.DataAccessManager;
import com.systex.jbranch.platform.common.dataaccess.query.QueryConditionIF;
import com.systex.jbranch.platform.common.errHandle.JBranchException;
import com.systex.jbranch.platform.server.info.FubonSystemVariableConsts;
import com.systex.jbranch.platform.util.IPrimitiveMap;
/**
 * 
 * Copy Right Information :  <br>
 * Project : fubon.wms<br>
 * JDK version used : JDK1.7.0 <br>
 * Description :Controller <br>
 * Comments Name : PMS314.java<br>
 * Author :Frank<br>
 * Date :2016年07月12日 <br>
 * Version : 1.01 <br>
 * Editor : Kevin<br>
 * Editor Date : 2017年01月31日<br>
 */

@Component("pms314")
@Scope("request")
public class PMS314 extends FubonWmsBizLogic {
	private DataAccessManager dam = null;
	private Logger logger = LoggerFactory.getLogger(PMS314.class);
	
    /*** 查詢資料 ***/
	public void inquire(Object body, IPrimitiveMap header) throws JBranchException {

		PMS314OutputVO outputVO = new PMS314OutputVO();
		dam = this.getDataAccessManager();
		
		QueryConditionIF condition = dam.getQueryCondition();
		SimpleDateFormat sdf=new SimpleDateFormat("yyyyMMdd"); 
		StringBuffer sql = new StringBuffer();    
		sql.append("SELECT * FROM "
				+ "TBPMS_SUMMIT_PAR "
				+ "WHERE SEQ = 1 ");
		condition.setQueryString(sql.toString());			
		outputVO.setResultList(dam.exeQuery(condition));
		sendRtnObject(outputVO);
				
	}
	
	/* ==== 【儲存】更新資料 ========*/
	public void save(Object body, IPrimitiveMap header) throws JBranchException {
//		Timestamp stamp = new Timestamp(System.currentTimeMillis());
		PMS314InputVO inputVO = (PMS314InputVO) body;
		DataAccessManager dam = this.getDataAccessManager();
	
		QueryConditionIF condition = dam.getQueryCondition();
		StringBuffer sql = new StringBuffer();   
		if(inputVO.getSeq() == 1){
			sql.append("UPDATE TBPMS_SUMMIT_PAR ");
			sql.append("SET FC_SUMMIT_INC = "+inputVO.getSummitINC()+", " );
			sql.append("FC_PEAK_INC = "+inputVO.getPeakINC()+", ");
			sql.append("SUMMIT_FC1 = '"+inputVO.getSummitFC1()+"', ");
			sql.append("SUMMIT_FC2 = '"+inputVO.getSummitFC2()+"', ");
			sql.append("SUMMIT_FC3 = '"+inputVO.getSummitFC3()+"', ");
			sql.append("SUMMIT_FC4 = '"+inputVO.getSummitFC4()+"', ");
			sql.append("SUMMIT_FC5 = '"+inputVO.getSummitFC5()+"', ");
			sql.append("PEAK_FC1 = '"+inputVO.getPeakFC1()+"', ");
			sql.append("PEAK_FC2 = '"+inputVO.getPeakFC2()+"', ");
			sql.append("PEAK_FC3 = '"+inputVO.getPeakFC3()+"', ");
			sql.append("PEAK_FC4 = '"+inputVO.getPeakFC4()+"', ");
			sql.append("PEAK_FC5 = '"+inputVO.getPeakFC5()+"', ");
			sql.append("FCH_1_INC = "+inputVO.getFch1INC()+", ");
			sql.append("FCH_2_INC = "+inputVO.getFch2INC()+", ");
			sql.append("FCH_AUM = "+inputVO.getFchAum()+", ");
			sql.append("FCH_NEW = "+inputVO.getFchNew()+", ");
			sql.append("MODIFIER = 'test', ");
			sql.append("LASTUPDATE = SYSDATE ");
			sql.append("WHERE SEQ = 1 ");			
		}else{
			sql.append("INSERT INTO TBPMS_SUMMIT_PAR ");
			sql.append("(FC_SUMMIT_INC, FC_PEAK_INC,  ");
			sql.append("SUMMIT_FC1, SUMMIT_FC2, SUMMIT_FC3, ");
			sql.append("SUMMIT_FC4, SUMMIT_FC5, ");
			sql.append("PEAK_FC1, PEAK_FC2, PEAK_FC3, ");
			sql.append("PEAK_FC4, PEAK_FC5, ");
			sql.append("FCH_1_INC, FCH_2_INC, ");
			sql.append("FCH_AUM, FCH_NEW, ");
			sql.append("CREATETIME, CREATOR, ");
			sql.append("MODIFIER, LASTUPDATE) ");
			sql.append("VALUES ( ");
			sql.append(inputVO.getSummitINC()+", ");
			sql.append(inputVO.getPeakINC()+", ");
			sql.append("'"+inputVO.getSummitFC1()+"', ");
			sql.append("'"+inputVO.getSummitFC2()+"', ");
			sql.append("'"+inputVO.getSummitFC3()+"', ");
			sql.append("'"+inputVO.getSummitFC4()+"', ");
			sql.append("'"+inputVO.getSummitFC5()+"', ");
			sql.append("'"+inputVO.getPeakFC1()+"', ");
			sql.append("'"+inputVO.getPeakFC2()+"', ");
			sql.append("'"+inputVO.getPeakFC3()+"', ");
			sql.append("'"+inputVO.getPeakFC4()+"', ");
			sql.append("'"+inputVO.getPeakFC5()+"', ");
			sql.append(inputVO.getFch1INC()+", ");
			sql.append(inputVO.getFch2INC()+", ");
			sql.append(inputVO.getFchAum()+", ");
			sql.append(inputVO.getFchNew()+", ");
			sql.append("SYSDATE, ");
			sql.append("'"
					+ (String)getUserVariable(FubonSystemVariableConsts.LOGINID)
					+ "', ");
			sql.append("'"
					+ (String)getUserVariable(FubonSystemVariableConsts.LOGINID)
					+ "', ");
			sql.append("SYSDATE) ");
		}
				
		condition.setQueryString(sql.toString());
		dam.exeUpdate(condition);
		sendRtnObject(null);
	}
	
	
	
}